commit 9545a15b07455e81474c414a0fd02c55d0ac2e85
Author: Xie Xingguo <xie.xingguo@zte.com.cn>
Date:   Tue Dec 18 19:54:23 2018 +0800

    mgr/DaemonServer: cancel upmap for merge source and target simultaneously
    
    Which is more effective...
    
    Signed-off-by: xie xingguo <xie.xingguo@zte.com.cn>

diff --git a/src/mgr/DaemonServer.cc b/src/mgr/DaemonServer.cc
index a1d5f87208e..5283bb4654d 100644
--- a/src/mgr/DaemonServer.cc
+++ b/src/mgr/DaemonServer.cc
@@ -2332,7 +2332,8 @@ void DaemonServer::adjust_pgs()
 		       << " has upmap" << dendl;
 	      upmaps_to_clear.insert(merge_target);
 	      ok = false;
-	    } else if (osdmap.have_pg_upmaps(merge_source)) {
+	    }
+            if (osdmap.have_pg_upmaps(merge_source)) {
 	      dout(10) << "pool " << i.first
 		       << " pg_num_target " << p.get_pg_num_target()
 		       << " pg_num " << p.get_pg_num()
